library/core/src/slice/mod.rs

```diff
@@ -559,15 +559,21 @@ impl<T> [T] {
             // Use the llvm.bswap intrinsic to reverse u8s in a usize
             let chunk = mem::size_of::<usize>();
             while i + chunk - 1 < ln / 2 {
-                // SAFETY: An unaligned u32 can be read from `i` if `i + 1 < ln`
-                // (and obviously `i < ln`), because each element is 2 bytes and
-                // we're reading 4.
+                // SAFETY: An unaligned usize can be read from `i` if `i + 1 < ln`
+                // (and obviously `i < ln`), because each element is 1 byte and
+                // we're reading 2.
+                //
                 // `i + chunk - 1 < ln / 2` # while condition
                 // `i + 2 - 1 < ln / 2`
                 // `i + 1 < ln / 2`
```

Comment:
This needs more updating for the `usize` case. We read `chunk` many bytes at once, not 2. And it's only safe to read if `i + chunk - 1 < ln`. And in the three lines below, we can't replace `chunk` by 2 anymore. Not quite sure how to best describe this here...

Reply:
I changed it, I think the proof I used is correct but I'm not sure I worded it clearly enough